app/locations/history.ts

```diff
@@ -39,9 +39,8 @@ export default class FragmentHistoryLocation extends HistoryLocation.extend(Guid
         if (fragment && newURL === currentPathAndQuery) {
             this.scrollToFragment.perform(fragment);
             return super.replaceURL(`${newURL}#${fragment}`);
-        } else {
-            return super.replaceURL(newURL);
         }
+        return super.replaceURL(newURL);
```

Comment:
Since this is happening in most places I imagine it's complaining about having an unneeded else when you're returning at the end of a if statement.

Reply:
Yep, [no-else-return](https://eslint.org/docs/rules/no-else-return), as above.